### PR TITLE
MinMdns - do not reset hasIP/hasHostPort on partial failure

### PR DESCRIPTION
**Pull request by andy31415**
MinMdns can receive several packets and if any of the packets are ok,
a single valid ip or host/port is sufficient.

In practice, this was failing on mdns on ipv6 only builds when both ipv6
and ipv4 were sent in order. Specifically:
  - MDNS packet contains AAAA followed by A
  - Resolver parses AAAA and sets a valid IP
  - Resolver sees A and fails to parse (no IPv4 support) and resets
    valid IP

As a result, dns resolution was never calling complete (because it
thinks no valid IPs).

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by andy31415** [comment]
Moving to draft temporarely until fully validated.

The changes to A/AAAA logic (for mHasIP) were validated, however our logic for mHasHostPort seems to potentially have some logic issues. Waiting to merge only once this is validated.

File: src/lib/dnssd/Resolver_ImplMinimalMdns.cpp [hunk 1]
```diff
@@ -321,9 +327,13 @@ void PacketDataReporter::OnResource(ResourceType type, const ResourceData & data
 
 void PacketDataReporter::OnComplete(ActiveResolveAttempts & activeAttempts)
 {
-    if ((mDiscoveryType == DiscoveryType::kCommissionableNode || mDiscoveryType == DiscoveryType::kCommissionerNode) &&
-        mDiscoveredNodeData.IsValid())
+    if (mDiscoveryType == DiscoveryType::kCommissionableNode || mDiscoveryType == DiscoveryType::kCommissionerNode)
     {
+        if (!mDiscoveredNodeData.IsValid())
+        {
+            ChipLogError(Discovery, "Discovered not data is not valid. Commissioning discovery not complete.");
```
Comment (mdamle) [inline src/lib/dnssd/Resolver_ImplMinimalMdns.cpp:334]:
```suggestion
            ChipLogError(Discovery, "Discovered node data is not valid. Commissioning discovery not complete.");
```

**Comment by andy31415** [comment]
Marking back as ready for review: I believe this is a net improvement over previous code.

**Comment by drempelg** [comment]
Yeah, verified that we need it to get anywhere with our DUT.  Came here to say +1

but it's already merged.

:)

**Pull request merged**